python/ctsm/param_utils/set_paramfile.py

```diff
@@ -0,0 +1,186 @@
+    if throw_error and not is_ndim_correct:
+        raise RuntimeError(f"Incorrect N dims: Expected {expected}, got {actual}")
+    return is_ndim_correct
+
+
+def main():
+    """
+    Main entry point for the script.
+    Parses arguments, opens a netCDF file, makes changes, and saves a new netCDF file.
+    """
+    args = get_arguments()
+
+    ds_in_masked_scaled = open_paramfile(args.input, mask_and_scale=True)
+
+    ds_in = open_paramfile(args.input)
+    ds_out = ds_in.copy()
+
+    # If --drop-other-pfts was given, drop PFTs not in args.pft
+    if args.drop_other_pfts:
+        pft_names = check_pfts_in_paramfile(args.pft, ds_out)
+        indices = get_selected_pft_indices(args.pft, pft_names)
+        ds_out = ds_out.isel({"pft": indices})
+
+    # If any variables specified, drop others
+    if args.variables:
+        vars_to_drop = []
+        for var in ds_in:
+            if var not in args.variables:
+                vars_to_drop.append(var)
+        ds_out = ds_out.drop_vars(vars_to_drop)
+
+    # Apply parameter changes, if any
+    for chg in args.param_changes:
+        var, new_value = chg.split("=")
+
+        # TODO: Add handling of multi-dimensional parameters
+        if ds_out[var].ndim > 1:
+            raise NotImplementedError("Can't yet change multi-dimensional parameters")
+
+        if "," in new_value:
+            new_value_list = new_value.split(",")
+            new_value = np.array(new_value_list)
+        else:
+            new_value = np.array(new_value)
+
+        # Some NaN handling isn't yet implemented
+        if np.any(np.char.lower(new_value) == "nan"):
+            # TODO: Add code to set integer variables to NaN (this might not be possible)
+            if is_integer(ds_in[var].values):
+                raise NotImplementedError(f"Not able to set NaN for integer parameters: {chg}")
+            # TODO: Add code to set integer variables to NaN (this might not be possible)
+            if "_FillValue" not in ds_in_masked_scaled[var].encoding:
+                raise NotImplementedError(
+                    f"Not able to set NaN if parameter doesn't already have fill value: {chg}"
+                )
+
+        new_value = new_value.astype(type(ds_out[var].dtype))
+
+        # Are we acting on just some PFTs? If so, we'll need some stuff.
+        just_some_pfts = PFTNAME_VAR in ds_out[var].coords and args.pft
+        if just_some_pfts:
+            pft_names = check_pfts_in_paramfile(args.pft, ds_out)
+            indices = get_selected_pft_indices(args.pft, pft_names)
+
+        # Check that correct number of dimensions were given for new values. Special handling needed
+        # if we're just acting on one PFT.
+        if just_some_pfts and len(args.pft) == 1:
+            check_correct_ndims(
+                ds_out[var].isel(pft=indices).squeeze(), new_value, throw_error=True
+            )
+        else:
+            check_correct_ndims(ds_out[var], new_value, throw_error=True)
```

Comment:
Simplify this to just have one `check_correct_ndims` call that uses `da_to_check = ...`.